pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -457,19 +457,20 @@ private Record readInput() {
     }
 
     @Override
-    public void close() {
+    synchronized public void close() {
```

Comment:
can you add a comment to why this method needs to be synchronized for future reference

Reply:
added comments